ui/Screen/ProjectCreation.svelte

```diff
@@ -214,6 +214,9 @@
 
     try {
       localState = await getLocalState(path);
+      if (!localState.branches.includes(defaultBranch)) {
```

Comment:
Would it make sense for `localState` to contain a `defaultBranch` instead of having the component set it to whatever branch exists at index 0?

Reply:
But that is what triggered the bug in the first place, because the defaultBranch is set to "master", so when a repository doesn't have that branch it leads to an error.

What happens here is, we get back the list of branches and "master" is not part of that list, so the dropdown was pre-selected to "null" instead of whatever is the first branch in the list.

Reply:
Good point. I think we should refactor this bit and move the `DEFAULT_BRANCH_FOR_NEW_PROJECTS` constant to the backend and then pass it via session or something. But I'd like to do this in a follow up.